internal/transport/proxy.go

```diff
@@ -115,32 +117,30 @@ func doHTTPConnectHandshake(ctx context.Context, conn net.Conn, backendAddr stri
 	return &bufConn{Conn: conn, r: r}, nil
 }
 
-// newProxyDialer returns a dialer that connects to proxy first if necessary.
-// The returned dialer checks if a proxy is necessary, dial to the proxy with the
-// provided dialer, does HTTP CONNECT handshake and returns the connection.
-func newProxyDialer(dialer func(context.Context, string) (net.Conn, error)) func(context.Context, string) (net.Conn, error) {
-	return func(ctx context.Context, addr string) (conn net.Conn, err error) {
-		var newAddr string
-		proxyURL, err := mapAddress(ctx, addr)
-		if err != nil {
-			if err != errDisabled {
-				return nil, err
-			}
-			newAddr = addr
-		} else {
-			newAddr = proxyURL.Host
+// proxyDial dials, connecting to a proxy first if necessary. Checks if a proxy
+// is necessary, dial to the proxy with using proxyDialer, does HTTP CONNECT
+// handshake, and returns the connection.
+func proxyDial(ctx context.Context, addr string, grpcUA string) (conn net.Conn, err error) {
+	var newAddr string
+	proxyURL, err := mapAddress(ctx, addr)
+	if err != nil {
+		if err != errDisabled {
```

Comment:
As we had discussed offline, please delete `errDisabled` and unfragment the handling of errors.

Reply:
Made this change.